src/solver/conv_asm_7x7c3h224w224k64u2v2p3q3f1.cpp

```diff
@@ -65,17 +68,11 @@ bool ConvAsm7x7c3h224w224k64u2v2p3q3f1::IsApplicable(const ExecutionContext& ctx
 #endif
     if(!(name == "gfx800" || name == "gfx802" || name == "gfx803" || name == "gfx804" ||
          name == "gfx900" || name == "gfx904" || name == "gfx906" || name == "gfx908"))
-    {
         return false;
-    }
```

Comment:
Curly braces are needed here. There was some misunderstanding here, I meant that curly braces are not required where the condition is short enough and the returned statement occupies one line. My comment related primarily to the new code. I should have explained this in more detail.

Reply:
OK, this is quite arbitrary and I find it hard to agree with. The language allows two choices: (1) no braces when the conditional has a single statement in the body and that has nothing to do with the length of the condition. That can be a source of bugs in absence of formatting, so I use `{}` all the time (choice #2). What you're asking is an arbitrary style choice and I find it inconsistent. Should we define how long is a "long condition"?

Reply:
I think it must be formalized and forced by clang-format.
If it's not forced by clang-format, we are free to do whatever we want - and most probably we want to argue what's the best codestyle ever.
@atamazov @averinevg please add appropriate check for it and we automatically will follow. I see no point to check codestyle on review.

Reply:
I'll see if I can set up a clang format for this

Reply:
I'm of the opinion that we should always use `{}` and modify `.clang-format` to put the starting `{` on the same line as `if` so that code height remains somewhat small.

On the general topic of style, I've opened up https://github.com/ROCmSoftwarePlatform/MIOpen/issues/2482 for discussion on adopting Google Style Guide.

Reply:
@CAHEK7 
> If it's not forced by clang-format, we are free to do whatever we want

NO. If there is no enforcing tool, then code review comes into play. That's one of the intents of code reviews.

The reviewers are spending time, and **_good authors value this and learn from the reviews_**. And professional engineers are able to adapt his/her own tastes to whatever practices used in the communities (they joined).

@amberhassaan 
Is it fine that a small code fix you were asked to make here transformed to a separate ticket, discussion thread here and additional work for Evgenii?

Reply:
@atamazov NO for you NO)
The problem happened because in the first comment it was recommended to remove the brackets for single statement conditions (which was done) and later for some of them it was recommended to return the brackets if the condition is long enough. It is too fuzzy and it's a matter of taste, not a rule.

clang-format can handle curly braced case in one way or another. Probably it handles it not ideally, but in uniform and automated way, which simplify everything. So, it must be used, moreover, if the ideal rules can't be formalized and each reviewer may have his own flavor of ideal solution.
And as you've already said - we must adapt our tastes to make the life easier.

Just to mention, I'm talking about the case which can be handled by clang-format.
If it can't handle the case with includes (quotes vs brackets), so I don't ask for it, but @pfultz2 provided a reference how to properly fix it.
If it can't handle naming convention, I don't ask for it, I'm asking for some sort of small notes like "`WeUseCamelCase` for types and classes and `we_use_underscores` for variables". It would help a lot for newcomers or if the code is mess.
Since it's an opensource project and available for everyone, it would be nice to have such a document.

And as I've already said - we want to argue about the codestyle, it happens over and over again.

Reply:
@atamazov : no, it's not fine that we are discussing style issues that would've been easily handled if we had a style document. We are creating unnecessary back and forth for minor matters. We need to have a `developer's guide` document in the repository that documents all the nuances of things like `Invoker & InvokerFactory` and how `in is out and out is in` in backward convolution and style issues like include files and variable/function naming etc. Hence the discussion in #2483